### PR TITLE
Updates to PubWise Analytics - Enable config, UTM & Debug/Logging Upd…

### PR DESCRIPTION
**Pull request by GLStephen**
## Type of change

This is an update to PubWise.io Analytics Adapter to support UTM parameters and improve loading when enableAnalytics is called in all supported places.

### UTM Tracking

PubWise analytics now tracks UTM parameters and allows segments to be built on campaign in the PubWise UI, this is the foundation for additional functionality in the "Pro" version of PubWise analytics that will allow advanced segmentation

## Updates

- Refactoring to improve behavior related to configuration and "enableAnalytics" call.
- Improvement to client side debugging capabilities, standardization of log prepend to filter console
- Improvement to how configuration is handled
- bump to 2.2

### CODE REVIEW

**Comment by GLStephen** [comment]
@matthewlane @protonate @mkendall07 Not to be pushy, but if this could make it into a 0.26.x patch that is upcoming I would be grateful and I can send cookies and cupcakes. Either way, appreciate it!

**Comment by GLStephen** [comment]
Merge conflicts have been resolved.

File: modules/pubwiseAnalyticsAdapter.js [hunk 1]
```diff
@@ -10,44 +11,117 @@ const utils = require('src/utils');
...
-let target_url = 'https://staging.api.pubwise.io';
-let pw_version = '2.1.3';
+const analyticsName = 'PubWise Analytics: ';
+let defaultUrl = 'https://api.pubwise.io/api/v4/event/default/';
+let pubwiseVersion = '2.2';
+let configOptions = {site: '', endpoint: 'https://api.pubwise.io/api/v4/event/default/', debug: ''};
+let pwAnalyticsEnabled = false;
+let utmKeys = {'utm_source': '', 'utm_medium': '', 'utm_campaign': '', 'utm_term': '', 'utm_content': ''};
+
+let getParameterByName = function (name) {
```
Comment (jaiminpanchal27) [inline modules/pubwiseAnalyticsAdapter.js]: This function is already defined in utils. Use that one.

File: modules/pubwiseAnalyticsAdapter.js [hunk 2]
```diff
@@ -10,44 +11,117 @@ const utils = require('src/utils');
...
+      }
+    }
+  } catch (e) {
+    dataBag['error_utm'] = 1;
+  }
+  return dataBag;
+}
+
+function sendEvent(eventType, data) {
+  utils.logInfo(analyticsName + 'Event ' + eventType + ' ' + pwAnalyticsEnabled, data);
```
Comment (jaiminpanchal27) [inline modules/pubwiseAnalyticsAdapter.js]: Use backticks here and other places in this file.

File: modules/pubwiseAnalyticsAdapter.js [hunk 3]
```diff
@@ -10,44 +11,117 @@ const utils = require('src/utils');
...
+
+function markEnabled() {
+  utils.logInfo(analyticsName + 'Enabled', configOptions);
+  pwAnalyticsEnabled = true;
+}
+
+function enrichWithMetrics(dataBag) {
+  try {
+    dataBag['pw_version'] = pubwiseVersion;
+    dataBag['pbjs_version'] = pbjs.version;
```
Comment (jaiminpanchal27) [inline modules/pubwiseAnalyticsAdapter.js]: Use $$PREBID_GLOBAL$$ here.

File: modules/pubwiseAnalyticsAdapter.js (hunk 1 above)
Comment (GLStephen) [inline modules/pubwiseAnalyticsAdapter.js]: Done

File: modules/pubwiseAnalyticsAdapter.js (hunk 3 above)
Comment (GLStephen) [inline modules/pubwiseAnalyticsAdapter.js]: Done

File: modules/pubwiseAnalyticsAdapter.js (hunk 2 above)
Comment (GLStephen) [inline modules/pubwiseAnalyticsAdapter.js]: Done

Comment (jaiminpanchal27) [inline modules/pubwiseAnalyticsAdapter.js]: Hey, I think you misunderstood me. I meant use template literals. 
Check here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

e.g: ``utils.logWarn(`Test string ${variable}.`);``

**Comment by GLStephen** [comment]
@jaiminpanchal27 I took another look and getParameterByName is not exported and is not used anywhere else currently

**Comment by jaiminpanchal27** [comment]
Oops.. Sorry for that. I will update utils and let you know. Also i replied to you on the backtick change.

**Comment by GLStephen** [comment]
@jaiminpanchal27 I pushed up a version with the required Uils change. If that will block us from getting into the next release then I'd rather duplicate the function. If you want me to pull your change later I can do that too. However, as it stands for the adpater to work I had to include that change in my update.

**Pull request merged**